datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/mutate/PatchEntityResolver.java

```diff
@@ -0,0 +1,159 @@
+            return new PatchEntityResult(urn, null, false, e.getMessage());
+          }
+        },
+        this.getClass().getSimpleName(),
+        "patchEntity");
+  }
+
+  private PatchEntityResult patchEntity(
+      @Nonnull PatchEntityInput input, @Nonnull QueryContext context) throws Exception {
+
+    // Handle URN resolution - either provided or auto-generated
+    final Urn entityUrn =
+        PatchResolverUtils.resolveEntityUrn(input.getUrn(), input.getEntityType());
+    final Authentication authentication = context.getAuthentication();
+
+    // Check authorization using GraphQL authorization pattern
+    if (!isAuthorized(input, context)) {
+      throw new AuthorizationException(
+          authentication.getActor().toUrnStr() + " is unauthorized to update entities.");
+    }
+
+    // Validate aspect exists
+    final AspectSpec aspectSpec =
+        _entityRegistry
+            .getEntitySpec(entityUrn.getEntityType())
+            .getAspectSpec(input.getAspectName());
+    if (aspectSpec == null) {
+      throw new IllegalArgumentException(
+          "Aspect "
+              + input.getAspectName()
+              + " not found for entity type "
+              + entityUrn.getEntityType());
+    }
+
+    // Create patch aspect using the same pattern as OpenAPI
+    final GenericAspect patchAspect =
+        PatchResolverUtils.createPatchAspect(
+            input.getPatch(), input.getArrayPrimaryKeys(), input.getForceGenericPatch(), context);
+
+    // Create MetadataChangeProposal
+    final com.linkedin.mxe.MetadataChangeProposal mcp =
+        PatchResolverUtils.createMetadataChangeProposal(
+            entityUrn,
+            input.getAspectName(),
+            patchAspect,
+            input.getSystemMetadata(),
+            input.getHeaders());
+
+    // Apply the patch
+    try {
+      Urn actor = UrnUtils.getUrn(authentication.getActor().toUrnStr());
+      IngestResult result =
+          _entityService.ingestProposal(
+              context.getOperationContext(),
+              mcp,
+              AuditStampUtils.createAuditStamp(actor.toString()),
+              false); // synchronous for GraphQL
+
+      if (result == null) {
+        return new PatchEntityResult(
+            entityUrn != null ? entityUrn.toString() : input.getUrn(),
+            null,
+            false,
+            "Failed to apply patch: ingestProposal returned null");
+      }
+
+      // Extract entity name from the patch operations
+      String entityName = PatchResolverUtils.extractEntityName(input.getPatch());
+
+      return new PatchEntityResult(entityUrn.toString(), entityName, true, null);
+    } catch (Exception e) {
+      return new PatchEntityResult(
+          entityUrn != null ? entityUrn.toString() : input.getUrn(),
+          null,
+          false,
+          "Failed to apply patch: " + e.getMessage());
+    }
+  }
+
+  private boolean isAuthorized(@Nonnull PatchEntityInput input, @Nonnull QueryContext context) {
```

Comment:
Same comments on the method above, the case where there is no urn is an invalid request.